include/libcyphal/transport/can/delegate.hpp

```diff
@@ -104,6 +105,45 @@ class TransportDelegate
 
     };  // CanardMemory
 
+    /// @brief Utility type with various helpers related to Canard AVL trees.
+    ///
+    /// @tparam Node Type of a concrete AVL tree node.
+    ///
+    template <typename Node>
+    struct CanardConcreteTree
+    {
+        /// @brief Recursively visits each node of the AVL tree, counting total number of nodes visited.
+        ///
+        /// Recursion goes first to the left child, then to the current node, and finally to the right child.
+        /// B/c AVL tree is balanced, the total complexity is `O(n)` and call stack depth
+        /// should not be deeper than `O(log(N))` (max 13 in case of 8192 active message subscriptions),
+        /// where `N` is the total number of tree nodes. Hence, the `NOLINT(misc-no-recursion)` exception.
+        ///
+        /// @tparam Visitor Type of the visitor callable.
+        /// @param node (sub-)root node of the AVL tree. Could be `nullptr`.
+        /// @param visitor The callable to be invoked for each non-null node.
+        /// @return Total count of visited nodes (including the `node` one). `0` if `node` is `nullptr`.
+        ///
+        template <typename Visitor>
+        static std::size_t visitCounting(CanardTreeNode* const node, Visitor&& visitor)  // NOLINT(misc-no-recursion)
+        {
+            if (node == nullptr)
+            {
+                return 0;
+            }
+
+            // Initial `1` is for the current node.
+            std::size_t count = 1;
+
+            count += visitCounting(node->lr[0], std::forward<Visitor>(visitor));
+            visitor(*reinterpret_cast<Node*>(node));
```

Comment:
It's worth adding that the reinterpret cast is unavoidable because we have to deal with a low-level and unsafe C API.

Reply:
It's actually in my next PR - where I added clang_tidy.